exonum-node/src/events/retries.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures::{TryFuture, TryFutureExt};
+use tokio::time::delay_for;
+
+use std::time::Duration;
+
+/// Retries the specified fallible future with the delay strategy specified by the `timeouts`
+/// iterator.
+pub async fn retry_future<Fut>(
```

Comment:
Why not just use `futures-retry` crate?

https://crates.io/crates/futures-retry